rfcs/20221024-persistent-cache-api-for-ocl-engine/README.md

```diff
@@ -0,0 +1,169 @@
+# Add Persistent Cache API for OpenCL Engine
+
+## Motivation
+In order to get all required information about GPU device oneDNN has to
+compile a dummy OpenCL kernel to use the compiler binary for querying that
+information.
+The kernel compilation happens during engine creation. In order to mitigate
+the associated overhead oneDNN caches the compiled binaries however, this
+doesn't help when the engine is created for the first time.
+
+OpenVINO reported that first engine creation overhead makes their network
+loading mechanism twice as slow.
+
+There is an option to rely on CL cache (implemented in driver). The main
+drawback of the option is that it requires to modify registry on Windows which
+doesn't work for some OpenVINO customers. Because of that OpenVINO has its own
+caching mechanism that caches serialized OpenCL kernels on disk.
+
+## Proposal
+
+oneDNN already provides persistent cache API for OpenCL runtime that allowed
+OpenVINO to eliminate first primitive creation overhead. OpenVINO can the same
+approach to eliminate first engine creation.
+
+The proposal is to introduce persistent cache API for engine. The API will be
+OpenCL runtime specific therefore it will reside in OpenCL interop namespace.
```

Comment:
Is it correct that it will replace two mechanisms? 1) engine creation and 2) mayiuse_ngen_kernel?

Reply:
Yes, cache blob will contain all necessary information to reduce engine creation overhead.